module-bsp/board/rt1051/puretx/bsp/battery_charger/battery_charger.cpp

```diff
@@ -26,6 +26,13 @@ namespace bsp::battery_charger
 
         const auto cfgFile              = purefs::dir::getUserDiskPath() / "batteryFuelGaugeConfig.cfg";
         constexpr auto registersToStore = 0xFF + 1;
+        constexpr auto configFileSizeWithoutChecksum =
+            registersToStore * sizeof(std::uint16_t); // each register has 2 bytes
+        constexpr auto configFileSizeWithChecksum =
+            configFileSizeWithoutChecksum + sizeof(std::uint16_t); // checksum has 2 extra bytes
+
+        static_assert(configFileSizeWithoutChecksum == 512, "Wrong config file size!");
+        static_assert(configFileSizeWithChecksum == 514, "Wrong config file size!");
```

Comment:
I thought it is a good practice. configFileSizeWithoutChecksum and configFileSizeWithChecksum based on uint16_t and it may change on others platforms.

Reply:
Well, `uint16_t` is guaranteed to be: unsigned 16-bit integer. It's a portable way of specifying platform-agnostic variables. The family of these aliases from `stdint.h` was added exactly for this purpose ;)

Reply:
Done